torch/csrc/jit/frontend/script_type_parser.cpp

```diff
@@ -162,13 +162,22 @@ c10::optional<std::string> ScriptTypeParser::parseBaseTypeName(
 
 TypePtr ScriptTypeParser::parseTypeFromExpr(const Expr& expr) const {
   if (expr.kind() == TK_SUBSCRIPT) {
+
+    if (resolver_) {
```

Comment:
I put this in as a hack at first with a hardcoded string passed in.  using expr.range().text() seems to be correct in most cases. I'm not sure how to decide if it's actually an OK approach or still a hack.